tokio/src/runtime/task/list.rs

```diff
@@ -91,13 +93,20 @@ impl<S: 'static> OwnedTasks<S> {
         task: T,
         scheduler: S,
         id: super::Id,
+        #[cfg(tokio_unstable)] spawned_at: &'static Location<'static>,
```

Comment:
Not a blocker.

Let `cfg` modify the parameter list is not a common pattern in Rust, I think we need to explore a better approach.

I don't think this issue blocks this PR, we could open another PR to simplify the implementation.

Reply:
My main feedback is also this. It may make sense to merge this into a struct with the id, or to wrap the location in something that can be zero-sized on stable, or to otherwise avoid conditional compilation everywhere this information is passed around.

Reply:
@Darksonn @ADD-SP what do you think of https://github.com/tokio-rs/tokio/pull/7417/commits/cf74e308b2ba3986b8fbd3a637cb43d97119bc8e? It did require adding `#[cfg(tokio_unstable)]` on the calls to the `spawn` hook, since they use the location which may now not be present if unstable isn't enabled. I felt like this was probably fine since the other hook functions are only called in cfg'd code anyway, but if that's an issue, I can change this to avoid that.